clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -841,6 +841,8 @@ raw2trace_t::process_marker_additionally(raw2trace_thread_data_t *tdata,
         log(4, "Rseq abort %d\n", tdata->rseq_past_end_);
         if (!adjust_and_emit_rseq_buffer(tdata, marker_val, marker_val))
             return false;
+        if (!append_delayed_branch(tdata, reinterpret_cast<app_pc>(marker_val)))
```

Comment:
Should adjust_and_emit_rseq_buffer call this rather than all its callers having to remember to do so?

Reply:
In append_bb_entries, the call to append_delayed_branch depends on whether the instruction is a branch or not, i.e. line 1961

           if (tdata->rseq_past_end_) {
                if (!adjust_and_emit_rseq_buffer(tdata, instr_pc))
                    return false;
                if (!instr->is_cti()) {
                    // Write out delayed branches now that we have a target.
                    if (!append_delayed_branch(tdata, orig_pc))
                        return false;
                }

which is difficult to generalize and add to adjust_and_emit_rseq_buffer.